addons/project/models/project_task.py

```diff
@@ -797,9 +799,15 @@ def copy_data(self, default=None):
             'dependent_ids': False,
         })
         vals_list = super().copy_data(default=default)
-        not_project_user = not self.env.user.has_group('project.group_project_user')
-        if not_project_user:
-            vals_list = [{k: v for k, v in vals.items() if k in self.SELF_READABLE_FIELDS} for vals in vals_list]
+        # filter only readable fields
+        vals_list = [
+            {
+                k: v
+                for k, v in vals.items()
+                if self._has_field_access(self._fields[k], 'read')
```

Comment:
Is `self._has_field_access()` dependent on `self`?  If that is the case, you must use the record corresponding to `vals` instead of `self`.

Reply:
It is not dependent on self. (only res.users depends on self because of the hacks it contains)